mypy/meet.py

```diff
@@ -161,11 +161,6 @@ def _is_overlapping_types(left: Type, right: Type) -> bool:
     if isinstance(left, PartialType) or isinstance(right, PartialType):
         assert False, "Unexpectedly encountered partial type"
 
-    # Ignore the error about using get_proper_type().
-    if isinstance(left, TypeGuardType) or isinstance(right, TypeGuardType):  # type: ignore[misc]
-        # A type guard forces the new type even if it doesn't overlap the old.
-        return True
```

Comment:
Why do you removing this part? It seems to me there may be cases when `narrow_declared_type()` will not even be called if types are not overlapping.